core/src/main/java/bisq/core/api/CorePriceService.java

```diff
@@ -65,6 +82,44 @@ public double getMarketPrice(String currencyCode) throws ExecutionException, Int
                 .collect(Collectors.toList());
     }
 
+    /** 
+     *  @return Data for market depth chart  
+    */
+    public MarketDepthInfo getMarketDepth(String currencyCode) {        
+        List<OpenOffer> buyOffers = openOfferManager.getOpenOffersByCurrency(Direction.BUY.name(), currencyCode); 
+        List<OpenOffer> sellOffers = openOfferManager.getOpenOffersByCurrency(Direction.SELL.name(), currencyCode); 
+
+        // Create buyer hashmap {key:price, value:count}
+        HashMap<Double,Long> buyHM = new HashMap<Double,Long>();        
+        buyOffers.forEach((openOffer) -> {
+            Double price = openOffer.getOffer().getOfferPayload().getCoinAmount();
+            Long previousValue = buyHM.get(price) != null ? buyHM.get(price) : Long.valueOf(0);
+            buyHM.put(price, previousValue + 1);
```

Comment:
do we want number of offers here? Or sum of volume? I haven't check how is it done in the desktop gui

Reply:
Oh you might be correct, I'm new to all the logic here.  When we call "getVolume" on an offer I'm not sure what that variable means exactly.  

Do you know how I could find the desktop code already present?  I never actually found where that was located.